v3_async_wip/v3_async_wip/iothub_mqtt_client.py

```diff
@@ -214,23 +199,46 @@ async def _keep_credentials_fresh(self) -> None:
                 logger.error("No SasTokenProvider. Cannot update credentials")
                 break
 
-    async def shutdown(self) -> None:
-        """Shut down the client.
+    async def start(self) -> None:
```

Comment:
>start [](http://example.com/codeflow?start=14&length=5)

start and stop are both idempotent. Is this by coincidence or by design? At the very least, these should be documented as "only call once" or "can be called multiple times". I'm not sure if there's any value in doing more than this (like adding artificial enforcement of non-idempotency).

Reply:
Have formalized docs and added tests to illustrate the by-design-ness of this.